Implemented Z85 Encoding

manzdagratiano opened the pull request:
The implementation follows the ZeroMQ RFC spec:
http://rfc.zeromq.org/spec:32/Z85/

Test Plan:
Added test cases for Z85 from the RFC and verified that they pass.

manzdagratiano commented:
Hey Folks! I'm using SJCL in my own project and I have the need to support Z85 encoding, so I wanted to add that implementation to SJCL. I've basically followed the reference C++ implementation at:
https://github.com/zeromq/rfc/blob/master/src/spec_32.c
while ensuring the encode/decode process works with bitArrays. Pray review and critique!

Nilos commented on test/z85_vectors.js:
```diff
@@ -0,0 +1,9 @@
+ */
+sjcl.test.vector.z85 =
+[
+[[0x864FD26F, 0xB559F75B], "HelloWorld"]
```
Please use `sjcl.codec.hex.toBits("0x864FD26FB559F75B")` so we have a full test here (you can also use the string here and transform it to bits in the test itself. This way even if the bitarray implementation changes, this test would still work (or better fail if necessary).

Nilos replied:
More tests than 1 would also be nice!

Nilos commented on config.mk:
```diff
@@ -1,2 +1,2 @@
-SOURCES= core/sjcl.js core/aes.js core/bitArray.js core/codecString.js core/codecHex.js core/codecBase32.js core/codecBase64.js core/sha256.js core/ccm.js core/ocb2.js core/gcm.js core/hmac.js core/pbkdf2.js core/random.js core/convenience.js 
+SOURCES= core/sjcl.js core/aes.js core/bitArray.js core/codecString.js core/codecHex.js core/codecBase32.js core/codecBase64.js core/codecZ85.js core/sha256.js core/ccm.js core/ocb2.js core/gcm.js core/hmac.js core/pbkdf2.js core/random.js core/convenience.js 
```
I think most people will not need z85 so please remove it from the default make config!

Nilos commented:
You also need to change the configure script and add z85 and its dependencies there!

manzdagratiano commented:
Hi Nils,
* The tests now encode from Hex string to bitArray to Z85, which decouples them from bitArray internals.
* Added codecZ85 and its dependencies (bitArray) to the configure script
* Removed codecZ85 from config.mk so it's not enabled by default (verified by running ```./configure``` without any arguments, and then with ```./configure --with-codecZ85```
* Added more test cases by scouring the internetz - PyZMQ as well as the Wikipedia Ascii85 example (from Thomas Hobbes' "Leviathan") verified through PyZMQ (The encoding shown on the Wikipedia page "https://en.wikipedia.org/wiki/Ascii85" is different from the Z85 encoding of the same text, since the character sets and their ordering are different from the btoa version).

Please see if I missed anything!

Nilos commented:
👍  Looks good! Thanks for your work!
